framework/src/modules/chain/bft/bft.js

```diff
@@ -151,11 +152,44 @@ class BFT extends EventEmitter {
 	/**
 	 * Computes maxHeightPreviouslyForged and prevotedConfirmedUptoHeight properties that are necessary
 	 * for creating a new block
-	 * @param delegatePubKey
+	 * @param delegatePublicKey
 	 * @return {Promise<{prevotedConfirmedUptoHeight: number, maxHeightPreviouslyForged: (number|*)}>}
 	 */
-	async computeBFTHeaderProperties(delegatePubKey) {
-		return this.finalityManager.computeBFTHeaderProperties(delegatePubKey);
+	async computeBFTHeaderProperties(delegatePublicKey) {
+		const previouslyForged = await this._getPreviouslyForgedMap();
+		const maxHeightPreviouslyForged = previouslyForged[delegatePublicKey] || 0;
+
+		return {
+			// prevotedConfirmedUptoHeight is up till height - 1
+			prevotedConfirmedUptoHeight: this.finalityManager.prevotedConfirmedHeight,
+			maxHeightPreviouslyForged,
+		};
+	}
+
+	/**
+	 * Saving a height which delegate last forged. this needs to be saved before broadcasting
+	 * so it needs to be outside of the DB transaction
+	 * @param delegatePublicKey
+	 * @param height
+	 */
+	async saveMaxHeightPreviouslyForged(delegatePublicKey, height) {
+		const previouslyForged = await this._getPreviouslyForgedMap();
+		const updatedPreviouslyForged = {
+			...previouslyForged,
+			[delegatePublicKey]: height,
```

Comment:
Don't we need to check if `previouslyForged[delegatePublicKey].height` is less than `height` before updating?

Reply:
I think it should always update to the last forging block regardless of height

Reply:
@lsilvs You are right. It's maxPreviouslyForgedHeight so it should check (confirmed with research team)